### PR TITLE
Demo parsing and correction of a 2 EDP Origin Report

### PR DESCRIPTION
**Pull request by uakyol**

### CODE REVIEW

File: dp_consistency/src/correctoriginreport.py
Comment (rmirisola) [inline dp_consistency/src/correctoriginreport.py:1]: Do you mind moving this to the report/ package, or to is own new package.

File: dp_consistency/src/correctoriginreport.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Read an Excel file.")
+    parser.add_argument(
+        "--path_to_report", required=True, help="Path to the Excel file"
+    )
+    args = parser.parse_args()
+
+    (measurements, excel) = readExcel(args.path_to_report)
```
Comment (rmirisola) [inline dp_consistency/src/correctoriginreport.py:155]: Can we add a unit test for that include parsing and correcting a given excel file?

File: dp_consistency/src/correctoriginreport.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#   7. The issue with the Linear TV shouldn't be corrected can be addressed by setting the variance of its measurements to 0.
+
+
+# Important Note: If sigma is set to a low number like 1, the corrected report returns numbers very close to 0. 
+SIGMA = 100
+
+AMI_FILTER = "AMI"
+MRC_FILTER = "MRC"
+
+EDP_ONE = "Google"
```
Comment (rmirisola) [inline dp_consistency/src/correctoriginreport.py:24]: Let's add a TODO to read EDP names dynamically from the file, and to make this work for more than 2 EDPs.

File: dp_consistency/src/correctoriginreport.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        measurements = {}
+        dfs = pd.read_excel(excel_file_path, sheet_name=None)
+        for edp in CUML_REACH_MAP:
+            sheet_name = CUML_REACH_MAP[edp]["sheet"]
+            df = dfs[sheet_name]
+            ami_rows = df[df[FILTER_COL_NAME] == AMI_FILTER]
+            mrc_rows = df[df[FILTER_COL_NAME] == MRC_FILTER]
+            # These rows are already sorted by timestamp.
+            ami_measurements = [
+                Measurement(measured_value, SIGMA)
```
Comment (rmirisola) [inline dp_consistency/src/correctoriginreport.py:55]: Note that for EDP = "Linear TV" sigma should be set to 0 (they don't use DP)

File: dp_consistency/.DS_Store
Comment (rmirisola) [inline dp_consistency/.DS_Store:1]: Is this intended to be checked in?

**Review by rmirisola — CHANGES_REQUESTED** [review]

File: dp_consistency/src/correctoriginreport.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
+from src.noiseninja.noised_measurements import Measurement
+from src.report.report import Report, MetricReport
+import sys
+import argparse
+import pandas as pd
+from functools import partial
+
+# This is a demo script that has the following assumptions :
+#   1. There are 2 EDPs one with Name Google, the other Linear TV.
+#   2. CUSTOM filter is ignored since it won't be used for the forseeable future.
```
Comment (rmirisola) [inline dp_consistency/src/correctoriginreport.py:10]: This is not necessarily true. Let's just state that CUSTOM filters are not yet supported in this tool.

File: dp_consistency/src/correctoriginreport.py
Comment (uakyol) [reply dp_consistency/src/correctoriginreport.py:1]: done

File: dp_consistency/src/correctoriginreport.py (hunk 2 above)
Comment (uakyol) [reply dp_consistency/src/correctoriginreport.py]: done

File: dp_consistency/.DS_Store
Comment (uakyol) [reply dp_consistency/.DS_Store:1]: thanks

File: dp_consistency/src/noiseninja/solver.py [hunk 5]
```diff
@@ -6,11 +6,13 @@
 
 from src.noiseninja.noised_measurements import SetMeasurementsSpec
 
-SOLVER = "clarabel"
+SOLVER = "highs"
```
Comment (rmirisola) [inline dp_consistency/src/noiseninja/solver.py:9]: intended? and change below?

**Review by rmirisola — APPROVED** [review]
Approved, just make sure changes in solver.py are intended before submitting.

**Pull request merged**